### ISSUE TITLE
Check avro fields and KSQL schema compatibility in CREATE ddl statements

### ISSUE THREAD

**Issue by hjafarpour** [confluentinc/ksql#540]
For avro format when user creates a stream or table using `CREATE STREAM` and `CREATE TABLE` statements check missing and incompatible fields in avro schema.

**Comment by joewood** [comment]
A suggestion for consistency - could the schema be checked and kept in sync regardless of the format (i.e. for JSON too)?

**Comment by hjafarpour** [comment]
@joewood Stream/Table schemas are always kept. This issue is for matching KSQL schema and the underlying topic's avro schema if it has avro format.

**Comment by joewood** [comment]
@hjafarpour right, but I mean if there's a registered schema in the registry, but it isn't necessary because the format is JSON. Could the KSQL schema and the schema registry still be validated? We could use the schema registry as a convenience, even if the encoding format isn't Avro.
